libsweep/src/sweep.cc

```diff
@@ -19,13 +20,25 @@ struct sweep_error {
 
 #define SWEEP_MAX_SAMPLES 4096
 
+struct sample {
+  int32_t angle;           // in millidegrees
+  int32_t distance;        // in cm
+  int32_t signal_strength; // range 0:255
+};
+
 struct sweep_scan {
-  int32_t angle[SWEEP_MAX_SAMPLES];           // in millidegrees
-  int32_t distance[SWEEP_MAX_SAMPLES];        // in cm
-  int32_t signal_strength[SWEEP_MAX_SAMPLES]; // range 0:255
+  sample samples[SWEEP_MAX_SAMPLES];
   int32_t count;
 };
 
+static sample parse_payload(const sweep::protocol::response_scan_packet_s& msg) {
+  sample ret;
+  ret.angle = msg.get_angle_millideg();
+  ret.distance = msg.distance;
+  ret.signal_strength = msg.signal_strength;
+  return ret;
```

Comment:
Should there be `SWEEP_ASSERT`s here to make sure
- signal strength is 0-255
- angle is 0-360 degree
- distance is positive